src/udev/udev-builtin-net_setup_link.c

```diff
@@ -50,17 +50,18 @@ static int builtin_net_setup_link(struct udev_device *dev, int argc, char **argv
                 }
         }
 
-        r = link_config_apply(ctx, link, dev, &name);
+        name = link_config_get_ifname(ctx, link, dev);
+        if (name)
+                udev_builtin_add_property(dev, test, "ID_NET_NAME", name);
+
+        r = link_config_apply(ctx, link, dev);
         if (r < 0) {
                 log_error_errno(r, "Could not apply link config to %s: %m", udev_device_get_sysname(dev));
                 return EXIT_FAILURE;
```

Comment:
How about simply turning this into `log_warning_errno()` and not returning (and dropping the rest of the patch)?

Reply:
That would work. On the other hand, I think the PR overall makes the code more readable. But if you strictly insist I can drop it and replace it w/ oneliner.